pkg/model/components/docker.go

```diff
@@ -69,6 +70,11 @@ func (b *DockerOptionsBuilder) BuildOptions(o interface{}) error {
 		clusterSpec.Docker.Version = &dockerVersion
 	}
 
+	dockerVersion, _ := semver.Parse(fi.StringValue(clusterSpec.Docker.Version))
```

Comment:
So docker versioning is no longer semver compliant. For example 17.06.01 (Leading 0s are not allowed). Also this does not take into account the new Docker versioning 17.06.01 (Year.Month.Release).

I'll leave the discussion of weather we want to set this as the default or not up to @justinsb